app/domain/invoice/item_evaluation.rb

```diff
@@ -111,16 +109,32 @@ def count(name, account, cost_center)
     invoice_items.first.count * amount_of_invoices
   end
 
-  def invoice_item_vats
-    # First convert the vat_rate to decimal,
-    # multiply with the total cost to get the absolute payed vats.
-    # Then sum those grouped by invoice item
-    payments_of_payed_invoices.sum('(IFNULL(vat_rate, 0) / 100) * (count * unit_cost)')
+  def invoice_item_vats(name, account, cost_center)
+    # Get the relevant invoices
+    relevant_invoice_ids = relevant_payments.of_fully_paid_invoices.payments.pluck(:invoice_id)
+
+    # Search invoice item which fits the identifiers and are attached to relevant payments
+    invoice_item = InvoiceItem.find_by(name: name,
+                                       account: account,
+                                       cost_center: cost_center,
+                                       invoice_id: relevant_invoice_ids)
+
+    # Invoice items with an empty vat_rate will be 0
+    return 0 unless invoice_item.vat_rate&.nonzero?
+
+    # We get the vat by multiplying the vat_rate with the paid amount excluding vat
+    invoice_item.vat_rate / 100 * amount_paid_without_vat(name, account, cost_center)
+  end
+
+  def amount_paid_without_vat(name, account, cost_center)
+    invoice_item = InvoiceItem.find_by(name: name, account: account, cost_center: cost_center)
+
+    count(name, account, cost_center) * invoice_item.unit_cost
```

Comment:
Was ist das für eine Logik-Änderung? Ich dachte du hast hier nur den Export hinzugefügt?

Reply:
Ja eigentlich schon. Das ist im ersten Commit gemacht weil ich beim Testdaten aufsetzen einen Bug entdeckt habe..
Hätte das + das Renaming eigentlich in einen eigenen PR packen können aber hab mir dann den Aufwand nicht gemacht.
Sorry für die Verwirrung :grin:

Reply:
Okay, macht Sinn. Aber was war denn der Bug? Beim Lesen des Changes wird mir nicht klar warum die vorherige Implementation falsch war.

Reply:
Das Problem ist dass die MwSt. auf die Rechnung und nicht auf die Zahlung berechnet werden soll.
Vorher wurde pro Zahlung die MwSt. (der gesamten zugehörigen Rechnung) berechnet und sumnmiert. Bei mehreren (> 1) Zahlungen pro Rechnungen führte dies dann zu einer falschen Berechnung.